src/compiler/program.ts

```diff
@@ -976,6 +976,22 @@ namespace ts {
 
         return program;
 
+        function resolveModuleNamesWorker(moduleNames: string[], containingFile: string, reusedNames?: string[], redirectedReference?: ResolvedProjectReference) {
+            performance.mark("beforeResolveModule");

```

Comment:
This works, but you could also do this:

```ts
// create once near the top of the file
const { enter: enterResolveModuleNames, exit: exitResolveModuleNames } = performance.createTimer(
    "ResolveModule",
    "beforeResolveModule",
    "afterResolveModule");
...
function resolveModuleNamesWorker(...) {
  enterResolveModuleNames();
  const result = actualResolveModuleNamesWorker(...);
  exitResolveModuleNames();
  return result;
}
```

You can also use `performance.createTimerIf` if you want to create a performance timer conditionally (such as if you only care about the metric when `--extendedDiagnostics` is set.

Whether you want to use `createTimer` is up to you.